src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -202,6 +202,12 @@ private static Assembly ResolveForUnknownRequestor(string requestedAssemblyName)
                     foreach (string loadedAssemblyFullPath in s_assembliesFromFiles.Keys)
                     {
                         string directoryPath = Path.GetDirectoryName(loadedAssemblyFullPath);
+
+                        if (!string.IsNullOrEmpty(requestedAssemblyIdentity.CultureName))
+                        {
+                            directoryPath = Path.Combine(directoryPath, requestedAssemblyIdentity.CultureName);
```

Comment:
Should we overwrite the directory path only if this directory actually exists: `Path.Combine(directoryPath, requestedAssemblyIdentity.CultureName);` ? This we way we can fallback to the loadedAssembly directory if no such sub-folder exists.